src/include/reconcile.h

```diff
@@ -0,0 +1,317 @@
+	uint32_t  multi_next;
+	size_t	  multi_allocated;
+
+	/*
+	 * Root pages are written when wrapping up the reconciliation, remember
+	 * the image we're going to write.
+	 */
+	WT_ITEM *wrapup_checkpoint;
+	bool	 wrapup_checkpoint_compressed;
+
+	/*
+	 * We don't need to keep the 0th key around on internal pages, the
+	 * search code ignores them as nothing can sort less by definition.
+	 * There's some trickiness here, see the code for comments on how
+	 * these fields work.
+	 */
+	bool	cell_zero;		/* Row-store internal page 0th key */
+
+	/*
+	 * We calculate checksums to find previously written identical blocks,
+	 * but once a match fails during an eviction, there's no point trying
+	 * again.
+	 */
+	bool	evict_matching_checksum_failed;
+
+	/*
+	 * WT_REC_DICTIONARY --
+	 *	We optionally build a dictionary of values for leaf pages. Where
+	 * two value cells are identical, only write the value once, the second
+	 * and subsequent copies point to the original cell. The dictionary is
+	 * fixed size, but organized in a skip-list to make searches faster.
+	 */
+	struct __wt_rec_dictionary {
+		uint64_t hash;				/* Hash value */
+		uint32_t offset;			/* Matching cell */
+
+		u_int depth;				/* Skiplist */
+		WT_REC_DICTIONARY *next[0];
+	} **dictionary;					/* Dictionary */
+	u_int dictionary_next, dictionary_slots;	/* Next, max entries */
+							/* Skiplist head. */
+	WT_REC_DICTIONARY *dictionary_head[WT_SKIP_MAXDEPTH];
+
+	/*
+	 * WT_REC_KV--
+	 *	An on-page key/value item we're building.
+	 */
+	struct __wt_rec_kv {
+		WT_ITEM	 buf;		/* Data */
+		WT_CELL	 cell;		/* Cell and cell's length */
+		size_t cell_len;
+		size_t len;		/* Total length of cell + data */
+	} k, v;				/* Key/Value being built */
+
+	WT_ITEM *cur, _cur;		/* Key/Value being built */
+	WT_ITEM *last, _last;		/* Last key/value built */
+
+	bool key_pfx_compress;		/* If can prefix-compress next key */
+	bool key_pfx_compress_conf;	/* If prefix compression configured */
+	bool key_sfx_compress;		/* If can suffix-compress next key */
+	bool key_sfx_compress_conf;	/* If suffix compression configured */
+
+	bool is_bulk_load;		/* If it's a bulk load */
+
+	WT_SALVAGE_COOKIE *salvage;	/* If it's a salvage operation */
+
+	bool cache_write_lookaside;	/* Used the lookaside table */
+	bool cache_write_restore;	/* Used update/restoration */
+
+	uint32_t tested_ref_state;	/* Debugging information */
+
+	/*
+	 * XXX
+	 * In the case of a modified update, we may need a copy of the current
+	 * value as a set of bytes. We call back into the btree code using a
+	 * fake cursor to do that work. This a layering violation and fragile,
+	 * we need a better solution.
+	 */
+	WT_CURSOR_BTREE update_modify_cbt;
+};
+
+typedef struct {
+	WT_UPDATE *upd;			/* Update to write (or NULL) */
+
+					/* Transaction IDs, timestamps */
+	wt_timestamp_t start_ts, durable_ts, stop_ts;
+	uint64_t start_txn, stop_txn;
+
+	bool upd_saved;			/* Updates saved to list */
+
+} WT_UPDATE_SELECT;
```

Comment:
Minor nit: I don't find having all that whitespace (blank lines) in the struct is more readable. I suggest getting rid of the 3 blank lines.

Reply:
Sure, pushed in d29ac8f.